### PR TITLE
framework/st_things: use secure storage for certificate

### PR DESCRIPTION
**Pull request by an4967**
- enable secure storage
- save svr_db in secure storage
- enable artik device uuid
- save certificate only with der format

### CODE REVIEW

File: framework/src/st_things/things_stack/src/common/framework/things_security_manager.c [hunk 1]
```diff
@@ -172,7 +166,110 @@ const unsigned char OIC_SVR_DB_DOXM_PAYLOAD[] = {
 	0x65, 0xFF, 0xFF
 };
 
-FILE *server_fopen(const char *path, const char *mode)	// pkss
+#ifdef CONFIG_SUPPORT_FULL_SECURITY
+#define SEE_IOTIVITY_MAXSIZE (1024 * 8)
```
Comment (sunghan-chang) [inline framework/src/st_things/things_stack/src/common/framework/things_security_manager.c]: ```#define SEE_IOTIVITY_MAXSIZE``` is at see_api.h. Do you need to define this again with same name?

File: framework/src/st_things/things_stack/src/common/framework/things_security_manager.c [hunk 2]
```diff
@@ -34,20 +34,14 @@
 #include "ss_sha2.h"
 #include "oxmverifycommon.h"
 #include "oic_string.h"
+#include "credresource.h"
 #include "utlist.h"
 #include "aclresource.h"
 #include "srmutility.h"
+#include "mbedtls/base64.h"
```
Comment (sunghan-chang) [inline framework/src/st_things/things_stack/src/common/framework/things_security_manager.c]: Is this external/include/mbedtls/base64.h? If it is, let's use < >

File: framework/src/st_things/things_stack/src/common/framework/things_security_manager.c [hunk 3]
```diff
@@ -172,7 +166,110 @@ const unsigned char OIC_SVR_DB_DOXM_PAYLOAD[] = {
...
+		if (svr.len <= 10) {
+			return NULL;
+		}
+	} else if (strcmp(mode, "w") == 0 || strcmp(mode, "wb") == 0) {
+		svr.mode = SVR_WRITE;
+	} else {
+		svr.mode = SVR_IDLE;
+		return NULL;
+	}
+	return (FILE *) svr.data;
```
Comment (sunghan-chang) [inline framework/src/st_things/things_stack/src/common/framework/things_security_manager.c]: remove a space after typecasting

**Comment by an4967** [comment]
@sunghan-chang I applied your request.

**Comment by seinfra** [comment]
Code Rule Check OK.

File: framework/src/st_things/things_stack/src/common/framework/things_security_manager.c [hunk 4]
```diff
@@ -172,7 +169,109 @@ const unsigned char OIC_SVR_DB_DOXM_PAYLOAD[] = {
...
+	svr_mode_t mode;
+} svr;
+
+FILE *server_fopen(const char *filename, const char *mode)
+{
+	svr.ptr = svr.data;
+	if (strcmp(mode, "r") == 0 || strcmp(mode, "rb") == 0) {
+		see_read_iotivity_storage((unsigned char *)svr.data, SEE_IOTIVITY_MAXSIZE, &svr.len);
+		svr.mode = SVR_READ;
+		if (svr.len <= 10) {
```
Comment (pradeepns85) [inline framework/src/st_things/things_stack/src/common/framework/things_security_manager.c]: What's the significance of  this conditional check with 10 ? any comment would be appreciable to understand this logic

Comment (an4967) [reply framework/src/st_things/things_stack/src/common/framework/things_security_manager.c]: @pradeepns85 it was an unclear point. I changed it to distinct value.

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by sunghan-chang** [comment]
@an4967 Could you check who can review this logic change? Here is no activity.

**Pull request closed without merging**